### ISSUE TITLE
Problems with assigments inside switch expressions

### ISSUE THREAD

**Issue by emilioplatzer** [rillig/gobco#27]
I tried with the switch example in [A Tour of Go](https://go.dev/tour/flowcontrol/9). It seems that the assigment inside the switch is cousing problems (I tried moving it outside and it works). 

```go
	switch os := runtime.GOOS; os {
	case "darwin":
		fmt.Println("OS X.")
```

The output is:
```
PS C:\hecho\go\switch> gobco
# example/switch [example/switch.test]
.\switch.go:10:37: undefined: os      
FAIL    example/switch [build failed]
FAIL
exit status 2open C:\Users\abcd\AppData\Local\Temp\gobco-6ee1adac66cac296\gobco-counts.json: 
```

## Complete code:
```go
package main

import (
	"fmt"
	"runtime"
)

func main() {
	fmt.Print("Go runs on ")
	switch os := runtime.GOOS; os {
	case "darwin":
		fmt.Println("OS X.")
	case "linux":
		fmt.Println("Linux.")
	default:
		// freebsd, openbsd,
		// plan9, windows...
		fmt.Printf("%s.\n", os)
	}
}
```

**Comment by rillig** [comment]
Hello Emilio,

Thanks for you detailed bug report. I was able to reproduce it using gobco 1.0.1. About 3 weeks ago, I discovered the completely buggy instrumentation of the `switch` statement myself, wondered why nobody had reported it in the last few years, and fixed it.

Please try again with the `master` branch, that worked for me. It could be that there are a few edge cases left that don't work. I will add a comprehensive and systematic test suite later.

Best,
Roland

**Comment by rillig** [comment]
Fixed in gobco 1.0.2, released yesterday.

**Comment by emilioplatzer** [comment]
Tested ok!. Thanks!
